packages/gitgraph-core/src/user-api/gitgraph-user-api.ts

```diff
@@ -99,30 +105,43 @@ class GitgraphUserApi<TNode> {
     return this._graph.createBranch(args).getUserApi();
   }
 
+  /**
+   * Tag a specific commit.
+   *
+   * @param options Options of the tag
+   */
+  public tag(options: GitgraphTagOptions): this;
   /**
    * Tag a specific commit.
    *
    * @param name Name of the tag
    * @param ref Commit or branch name or commit hash
    */
   public tag(
-    name: string,
-    ref?: Commit<TNode> | Commit["hash"] | Branch["name"],
-  ): this {
+    name: GitgraphTagOptions["name"],
```

Comment:
Avoid duplicating the type between all signatures.

![think-about-it](https://user-images.githubusercontent.com/1094774/54731488-d5f61d80-4b64-11e9-96b4-1a86165e068a.gif)